### PR TITLE
Polish Azure.ResourceManager.Search and prepare for release

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/search/Azure.ResourceManager.Search/src/autorest.md [hunk 1]
```diff
@@ -16,6 +16,30 @@ skip-csproj: true
...
+  ResourceType: SearchServiceResourceType
+  ShareablePrivateLinkResourceProperties: ShareableSearchServicePrivateLinkResourceProperties
+  ShareablePrivateLinkResourceType: ShareableSearchServicePrivateLinkResourceType
+  SharedPrivateLinkResource: SharedSearchServicePrivateLinkResource
+  SharedPrivateLinkResourceProperties: SharedSearchServicePrivateLinkResourceProperties
+  SharedPrivateLinkResourceProperties.privateLinkResourceId: -|arm-id
+  SharedPrivateLinkResourceProperties.resourceRegion: -|azure-location
+  SharedPrivateLinkResourceProvisioningState: SharedSearchServicePrivateLinkResourceProvisioningState
+  SharedPrivateLinkResourceStatus: SharedSearchServicePrivateLinkResourceStatus
+  UnavailableNameReason: SearchServiceUnavailableNameReason
```
Comment (fengzhou-msft) [inline sdk/search/Azure.ResourceManager.Search/src/autorest.md:41]:
```suggestion
  UnavailableNameReason: SearchServiceNameUnavailableReason
```

**Review by fengzhou-msft — APPROVED** [review]
Approved with comments

File: sdk/search/Azure.ResourceManager.Search/src/autorest.md (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/search/Azure.ResourceManager.Search/src/autorest.md]: fixed

**Pull request merged**